extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -520,47 +505,37 @@ export class AmpStoryReaction extends AMP.BaseElement {
    * @private
    */
   handleSuccessfulDataRetrieval_(response) {
-    if (!(response && 'data' in response)) {
-      dev().error(
-        TAG,
-        `Invalid reaction response, expected { data: ReactionResponseType, ...} but received ${response}`
-      );
-      return;
-    }
-
-    this.responseData_ = response.data;
-
-    this.hasUserSelection_ = this.responseData_.hasUserResponded;
-    if (this.hasUserSelection_) {
-      this.updateReactionOnDataRetrieval_(response.data);
-    }
+    devAssert(
+      response && 'options' in response,
+      `Invalid reaction response, expected { data: ReactionResponseType, ...} but received ${response}`
+    );
+    const numOptions = this.rootEl_.querySelectorAll(
+      '.i-amphtml-story-reaction-option'
+    ).length;
+    this.updateReactionOnDataRetrieval_(
+      response['options'].splice(0, numOptions)
```

Comment:
Only keep the first N options (N= options on the frontend quiz). This ensures the percentages add up to 100 if the backend returns unused options.